src/Processors/Formats/Impl/Parquet/ParquetDataBuffer.h

```diff
@@ -54,11 +54,14 @@ class ParquetDataBuffer
         auto necessary_bytes = count * sizeof(ParquetType);
         checkAvaible(necessary_bytes);
 
-        const ParquetType* src = reinterpret_cast<const ParquetType*>(data);
-
         for (std::size_t i = 0; i < count; i++)
         {
-            dst[i] = static_cast<TValue>(src[i]);
+            ParquetType temp;
+            auto offset = i * sizeof(ParquetType);
+
+            // necessary to prevent memory alignment issues https://github.com/ClickHouse/ClickHouse/issues/74512#issuecomment-2587260001
+            memcpy(&temp, data + offset, sizeof(ParquetType));
```

Comment:
I can also just add `NO_SANITIZE_UNDEFINED` and keep the same function as before. What do you prefer @nickitat?

Reply:
I'd prefer the current solution, and I'd also suggest to replace this code with a call to `unalignedLoad`